1-Draft/RFCNNNN-Multi-Line-Continuation.md

```diff
@@ -0,0 +1,301 @@
+preferable if the closing closure did not have to be at the start of a line,
+since there is no need for it to be. It would simply have to be the first token
+on line to close the statement, allowing for indentation within scripts.
+
+This alternative also allows for blank lines to be used within the enclosures,
+such that Scenario 3 in @dragonwolf83's comment could be supported and written
+like this:
+
+```PowerShell
+New-ADUser @`
+    -Name 'Jack Robinson'
+    -GivenName 'Jack'
+    -Surname 'Robinson'
+    -SamAccountName 'J.Robinson'
+
+    -UserPrincipalName (
+        'J.Robinson@enterprise.com'
+    )
+
+   # Get the list of regions for where a user would reside in to put the user into the correct region
+    -Path (
+        $region = Get-Region -Name 'Jack Robinson'
+        "OU=Users,OU=$region,DC=enterprise,DC=com"
+    )
+
+    -AccountPassword (
+        Read-Host -AsSecureString 'Input Password'
+    )
+
+    -Enabled $true
+`@
+
+Get-ChildItem @`
+    $rootFolder
+    -File
+    -Filter '*.ps*1'
+`@
+```
+
+The best part is that it doesn't appear this syntax would introduce a breaking
+change at all.
+
+If people feel the backtick is still not visible enough here (and therefore not
+desirable for this purpose), we should keep the `@` portion of the enclosures
+so that we can avoid breaking changes, and simply replace the backtick with
+something else. For example, we could do this instead:
+
+```PowerShell
+Get-ChildItem @-
+    $rootFolder
+    -File
+    -Filter '*.ps*1'
+-@
+```
+
+Backticks offer the advantage of continuing what they represent in PowerShell
+already (line continuation), and they are syntactically very similar to here-
+strings when used with the `@` symbol. That last point could be seen as a
+disadvantage as well, because they may be visually harder to distinguish from
+a single-quoted here-string.
+
+The `@-|-@` alternative doesn't pick up on the backtick for line continuation,
+but it is visually unique and easy to see in a script. The `-` character could
+be seen as representing the line that makes up the statement as well.
+
+### Inline splatting
+
+There has also been some discussion about the idea of inline splatting, using a
+format like `-@{...}` or `-@(...)`. Inline splatting has also been discussed
+separately on [RFC0002: Generalized Splatting](https://github.com/PowerShell/PowerShell-RFC/blob/master/2-Draft-Accepted/RFC0002-Generalized-Splatting.md), but using the syntax `@@{...}` or
+`@@(...)`.
+
+Here is an example showing what that might look like:
+
+```PowerShell
+Get-ChildItem -@{
+    LiteralPath = $rootFolder
+    File = $true
+    Filter = '*.ps*1'
+}
+```
+
+Using inline splatting to be able to span a single command across multiple
+lines like this has several limitations, including:
+
+1. You cannot transition to/from the inline splatted syntax without a bunch of
+manual tweaks to the command (either converting parameter syntax into hashtable
+or array syntax or vice versa).
```

Comment:
Assuming we implement inline splatting, we believe that this could be implemented using a PSSA rule. E.g. it could say that any number of parameters or column width over some value could trigger a rule with an auto-format to turn the in-line list of params and their values into a splatted inline hashtable.

Reply:
That may be helpful, but some questions back to you about this:
1. What about folks who don't use PSSA?
1. Would that be in Visual Studio Code only? Or other tools?
1. Once you convert, then what? If someone wants to add a parameter, will they get the same Intellisense? In any PowerShell scripting tool, or only in Visual Studio Code?
1. Depending on the syntax you go with for inline splatting, I may want to avoid that entirely, so hopefully it's configurable. I'll use backticks over `@@{...}` anyday.

Reply:
Responded to elsewhere where you also mentioned a PSSA rule.